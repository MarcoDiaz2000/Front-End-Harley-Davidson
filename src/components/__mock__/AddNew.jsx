import React, { useState } from 'react';
import MotorcycleForm from './MotorcycleForm';

const MockAddNew = () => {
  const [newMotor, setNewMotor] = useState({
    name: '',
    description: '',
    deposit: 0,
    finance_fee: 0,
    option_to_purchase_fee: 0,
    total_amount_payable: 0,
    duration: 0,
  });

  const handleChange = (e) => {
    const { name, value } = e.target;
    setNewMotor({
      ...newMotor,
      [name]: value,
    });
  };

  const handleSubmit = (event) => {
    event.preventDefault();
  };

  return (
    <div className="flex justify-center bodyBg overflow-y-auto pt-12 pb-12 min-h-screen">
      <div className="bg-customBg p-5 rounded-lg w-full sm:w-1/2 mb-12 white-shadow flex-shrink-0">
        <h1 className="text-white text-3xl mb-4 font-roboto font-bold">Add New Motorcycle</h1>
        <MotorcycleForm
          newMotor={newMotor}
          handleChange={handleChange}
          handleSubmit={handleSubmit}
        />
      </div>
    </div>
  );
};
export default MockAddNew;
