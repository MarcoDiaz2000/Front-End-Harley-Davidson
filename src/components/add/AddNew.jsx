import React, { useState, useRef } from 'react';
import { useDispatch } from 'react-redux';
import { toast } from 'react-toastify';
import { createBike } from '../../redux/bikes/thunk';
import MotorcycleForm from './MotorcycleForm';

const AddNew = () => {
  const imageInputRef = useRef(null);
  const dispatch = useDispatch();
  const [newMotor, setNewMotor] = useState({
    name: '',
    description: '',
    deposit: 0,
    finance_fee: 0,
    option_to_purchase_fee: 0,
    total_amount_payable: 0,
    duration: 0,
  });

  const handleChange = (e) => {
    const { name, value } = e.target;
    setNewMotor({
      ...newMotor,
      [name]: value,
    });
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    if (!newMotor.name || newMotor.name.trim() === '') {
      toast.error('The name field is required.');
      return;
    }

    const data = new FormData();

    data.append('item[name]', event.target.name.value);
    data.append('item[image]', event.target.image.files[0]);
    data.append('item[description]', event.target.description.value);
    data.append('item[deposit]', event.target.deposit.value);
    data.append('item[finance_fee]', event.target.finance_fee.value);
    data.append('item[option_to_purchase_fee]', event.target.option_to_purchase_fee.value);
    data.append('item[total_amount_payable]', event.target.total_amount_payable.value);
    data.append('item[duration]', event.target.duration.value);
    dispatch(createBike({ data }));

    toast.success('New motorcycle successfully added');
    setNewMotor({
      name: '',
      description: '',
      deposit: 0,
      finance_fee: 0,
      option_to_purchase_fee: 0,
      total_amount_payable: 0,
      duration: 0,
    });
    if (imageInputRef.current) {
      imageInputRef.current.value = '';
    }
  };

  return (
    <div className="flex justify-center bodyBg overflow-y-auto pt-12 pb-12 min-h-screen">
      <div className="bg-customBg p-5 rounded-lg w-full sm:w-1/2 mb-12 white-shadow flex-shrink-0">
        <h1 className="text-white text-3xl mb-4 font-roboto font-bold">Add New Motorcycle</h1>
        <MotorcycleForm
          newMotor={newMotor}
          handleChange={handleChange}
          handleSubmit={handleSubmit}
          imageInputRef={imageInputRef}
        />
      </div>
    </div>
  );
};
export default AddNew;
