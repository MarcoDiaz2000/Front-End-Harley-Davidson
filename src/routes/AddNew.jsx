import { toast } from 'react-toastify';
import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { createMotorcycle } from '../redux/motorcycles/motorcycles';

export default function AddNew() {
  const dispatch = useDispatch();
  const [newMotor, setNewMotor] = useState({
    name: '',
    image: '',
    description: '',
    deposit: 0,
    finance_fee: 0,
    option_to_purchase_fee: 0,
    total_amount_payable: 0,
    duration: 0,
  });

  const handleChange = (e) => {
    const { name, value } = e.target;
    setNewMotor({
      ...newMotor,
      [name]: value,
    });
  };

  const handleSubmit = () => {
    dispatch(createMotorcycle(newMotor));
    toast.success('New motorcycle successfully added');
    setNewMotor({
      name: '',
      image: '',
      description: '',
      deposit: 0,
      finance_fee: 0,
      option_to_purchase_fee: 0,
      total_amount_payable: 0,
      duration: 0,
    });
  };

  return (
    <div className="flex justify-center items-center h-screen bg-gray-200">
      <div className="bg-green-500 p-6 rounded-lg w-1/2 max-h-[100vh] overflow-y-auto">
        <h1 className="text-white text-2xl mb-4">Add New Moto</h1>
        <form>
          <div className="mb-4">
            <label htmlFor="name" className="text-white block mb-2">
              Name:
              <input
                id="name"
                className="w-full p-2 rounded text-black"
                type="text"
                name="name"
                onChange={handleChange}
                value={newMotor.name}
              />
            </label>
          </div>
          <div className="mb-4">
            <label htmlFor="image" className="text-white block mb-2">
              Image URL:
              <input
                id="image"
                className="w-full p-2 rounded text-black"
                type="text"
                name="image"
                onChange={handleChange}
                value={newMotor.image}
              />
            </label>
          </div>
          <div className="mb-4">
            <label htmlFor="description" className="text-white block mb-2">
              Description:
              <textarea
                id="description"
                className="w-full p-2 rounded text-black"
                name="description"
                onChange={handleChange}
                value={newMotor.description}
              />
            </label>
          </div>
          <div className="mb-4">
            <label htmlFor="deposit" className="text-white block mb-2">
              Deposit:
              <input
                id="deposit"
                className="w-full p-2 rounded text-black"
                type="number"
                name="deposit"
                onChange={handleChange}
                value={newMotor.deposit}
              />
            </label>
          </div>
          <div className="mb-4">
            <label htmlFor="finance_fee" className="text-white block mb-2">
              Finance Fee:
              <input
                id="finance_fee"
                className="w-full p-2 rounded text-black"
                type="number"
                name="finance_fee"
                onChange={handleChange}
                value={newMotor.finance_fee}
              />
            </label>
          </div>
          <div className="mb-4">
            <label htmlFor="option_to_purchase_fee" className="text-white block mb-2">
              Option to Purchase Fee:
              <input
                id="option_to_purchase_fee"
                className="w-full p-2 rounded text-black"
                type="number"
                name="option_to_purchase_fee"
                onChange={handleChange}
                value={newMotor.option_to_purchase_fee}
              />
            </label>
          </div>
          <div className="mb-4">
            <label htmlFor="total_amount_payable" className="text-white block mb-2">
              Total Amount Payable:
              <input
                id="total_amount_payable"
                className="w-full p-2 rounded text-black"
                type="number"
                name="total_amount_payable"
                onChange={handleChange}
                value={newMotor.total_amount_payable}
              />
            </label>
          </div>
          <div className="mb-4">
            <label htmlFor="duration" className="text-white block mb-2">
              Duration:
              <input
                id="duration"
                className="w-full p-2 rounded text-black"
                type="number"
                name="duration"
                onChange={handleChange}
                value={newMotor.duration}
              />
            </label>
          </div>
          <button type="button" onClick={handleSubmit} className="bg-white text-green-500 p-2 rounded hover:bg-green-200">
            Add
          </button>
        </form>
      </div>
    </div>
  );
}
