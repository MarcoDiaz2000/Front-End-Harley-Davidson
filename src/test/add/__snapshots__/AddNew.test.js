import React from 'react';
import renderer from 'react-test-renderer';
import AddNew from '../../components/__mock__/AddNew';

describe('AddNew', () => {
  it('renders correctly', () => {
    const component = renderer.create(<AddNew />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
