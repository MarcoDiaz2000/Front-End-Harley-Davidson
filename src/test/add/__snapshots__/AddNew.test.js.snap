// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNew renders correctly 1`] = `
<div
  className="flex justify-center bodyBg overflow-y-auto pt-12 pb-12 min-h-screen"
>
  <div
    className="bg-customBg p-5 rounded-lg w-full sm:w-1/2 mb-12 white-shadow flex-shrink-0"
  >
    <h1
      className="text-white text-3xl mb-4 font-roboto font-bold"
    >
      Add New Motorcycle
    </h1>
    <form>
      <div
        className="mb-3"
      >
        <label
          className="text-white block mb-2 font-roboto font-bold"
          htmlFor="name"
        >
          Name *:
          <input
            className="w-full p-2 rounded text-black"
            id="name"
            name="name"
            type="text"
          />
        </label>
      </div>
      <div
        className="mb-3"
      >
        <label
          className="text-white block mb-2 font-roboto font-bold"
          htmlFor="image"
        >
          Image *:
          <input
            className="w-full p-2 rounded text-black font-roboto font-bold"
            id="image"
            name="image"
            type="file"
          />
        </label>
      </div>
      <div
        className="mb-3"
      >
        <label
          className="text-white block mb-2 font-roboto font-bold"
          htmlFor="description"
        >
          Description *:
          <input
            className="w-full p-2 rounded text-black font-roboto font-bold"
            id="description"
            name="description"
            type="text"
          />
        </label>
      </div>
      <div
        className="mb-3"
      >
        <label
          className="text-white block mb-2 font-roboto font-bold"
          htmlFor="deposit"
        >
          Deposit:
          <input
            className="w-full p-2 rounded text-black"
            id="deposit"
            name="deposit"
            type="number"
          />
        </label>
      </div>
      <div
        className="mb-3"
      >
        <label
          className="text-white block mb-2 font-roboto font-bold"
          htmlFor="finance_fee"
        >
          Finance Fee:
          <input
            className="w-full p-2 rounded text-black"
            id="finance_fee"
            name="finance_fee"
            type="number"
          />
        </label>
      </div>
      <div
        className="mb-3"
      >
        <label
          className="text-white block mb-2 font-roboto font-bold"
          htmlFor="option_to_purchase_fee"
        >
          Option to Purchase Fee:
          <input
            className="w-full p-2 rounded text-black"
            id="option_to_purchase_fee"
            name="option_to_purchase_fee"
            type="number"
          />
        </label>
      </div>
      <div
        className="mb-3"
      >
        <label
          className="text-white block mb-2 font-roboto font-bold"
          htmlFor="total_amount_payable"
        >
          Total Amount Payable:
          <input
            className="w-full p-2 rounded text-black"
            id="total_amount_payable"
            name="total_amount_payable"
            type="number"
          />
        </label>
      </div>
      <div
        className="mb-3"
      >
        <label
          className="text-white block mb-2 font-roboto font-bold"
          htmlFor="duration"
        >
          Duration:
          <input
            className="w-full p-2 rounded text-black"
            id="duration"
            name="duration"
            type="number"
          />
        </label>
      </div>
      <div
        className="flex justify-center"
      >
        <button
          className="bg-harleyOrange text-white py-3 px-12 rounded hover:bg-orange-700 shadow-md font-roboto font-bold"
          type="submit"
        >
          Add New Motorcycle
        </button>
      </div>
    </form>
  </div>
</div>
`;
